src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2268,6 +2283,11 @@ RunDistributedExecution(DistributedExecution *execution)
 			int eventCount = WaitEventSetWait(execution->waitEventSet, timeout, events,
 											  eventSetSize, WAIT_EVENT_CLIENT_READ);
 			ProcessWaitEvents(execution, events, eventCount, &cancellationReceived);
+			if (cancellationReceived)
+			{
+				/* user requested cancel, do not continue the execution */
+				break;
```

Comment:
why do we need this now, but not before?

Reply:
Previously this was on the `while()` condition, moved here just to simplify the `while` condition as now I added on more thing there:
> while (execution->unfinishedTaskCount > 0 && !cancellationReceived)

Reply:
got it, thought I might be missing something. I would have a slight preference for keeping it in the `while` just to see the 3 conditions in one place.

Reply:
makes sense, moved again